### ISSUE TITLE
No distinction between 'is' and 'are' leads to ambiguity errors

### ISSUE THREAD

**Issue by McEarl** [Naproche/Naproche-SAD#8]
I get an ambiguity error because Naproche does not distinguish between the words _is_ and _are_. Thus a statement which is not at all ambiguous for a human reader becomes ambiguous for Naproche.

Example:

`Definition. Let x be a set. Assume that every element of x is a set. The intersection of x is {z | For all elements y of x z is an element of y}.`

`Definition. Let x,y be sets. The intersection of x and y is {z | z is an element of x and z is an element of y}.`

`Lemma. Let x,y,z be sets. The intersection of x and y is an element of z.`

The lemma produces the following error message:

`ambiguity error[aElementOf(theIntersectionOfAnd(x,y),z).`
`,(aElementOf(theIntersectionOf(x),z) and aElementOf(y,z)).`
`]`

**Comment by anfelor** [comment]
I don't think this can be fixed in Naproche-SAD itself, but might be an interesting example for the rewrite. What do you think @adelon?

**Comment by adelon** [comment]
Yes, this is good example! I see two ways of fixing this:

* Use a 'longest match' rule for patterns. Thus `intersection of _ and _` would take precedence over `intersection of _`. This is already implemented in the rewrite.
* Require correct grammar (i.e. `are` instead of `is`) for lists of terms, as suggested. I'm slowly adding these sorts of requirements to the rewrite as well. 

For now, you can use parentheses for disambiguation:

```
(The intersection of x and y) is an element of z
```

**Comment by PeterKoepke** [comment]
Human parsing seems to pull every available string for disambiguation. 
So "The intersection of x and y is an element of z."  can be understood uniquely because of the singular form "is".
But what about
"Lemma. Let x,y,z,u be sets. The intersection of x and y and z are 
elements of u." 
There are no forms of "is" to distinguish a 2-plurality from a 3-plurality. This sentence is ambiguous for humans as well.
We should be very careful if we want to eliminate ambiguity by rules. Is "longest match" a natural rule for human disambiguation? Or is it just pragmatic and useful for formal languages? Obviously the two intersection operations considered are prone to ambiguity. So mathematicians use symbolic formalisms exactly to avoid this: whereas brackets in natural text are frowned upon, one can use symbols like \bigcap and \cap, and the problem with the "and" goes away.
I think that the ambiguity error encountered by McEarl is a helpful warning which should lead to better reformulations. Natural formalism does not mean to express everything in words.
